backend/src/main/scala/cromwell/backend/wdl/ReadLikeFunctions.scala

```diff
@@ -6,10 +6,28 @@ import wdl4s.expression.WdlStandardLibraryFunctions
 import wdl4s.parser.MemoryUnit
 import wdl4s.types.{WdlArrayType, WdlFileType, WdlObjectType, WdlStringType}
 import wdl4s.values._
+import cats._, implicits._
 
 import scala.util.{Failure, Success, Try}
 
-trait ReadLikeFunctions extends PathFactory { this: WdlStandardLibraryFunctions =>
+trait FileSizeLimitationConfig {
+  val readLinesLimit = 128000
+  val readBoolLimit = 5
+  val readIntLimit = 19
+  val readFloatLimit = 50
+  val readStringLimit = 128000
+  val readJsonLimit = 128000
+  val readTsvLimit = 128000
+  val readMapLimit = 128000
+  val readObjectLimit = 128000
+}
+
+case class FileSizeTooBig(override val getMessage: String) extends Exception
+
+trait ReadLikeFunctions extends PathFactory with FileSizeLimitationConfig { this: WdlStandardLibraryFunctions =>
+
+  def fileSize: WdlValue=> Try[Long] = 
+    w => Try(buildPath(w.valueString).size)
```

Comment:
per kristian on the bug: if file is larger than limit than should be an error, not read first X bytes and silently drop the rest.

I'm kind of hesitant to re-write `size` or invent something to avoid a metadata request, but perhaps @katevoss or @geoffjentry have opinions on this tradeoff?